intelmq/bots/parsers/twitter/parser.py

```diff
@@ -0,0 +1,90 @@
+"""
+
+Parser of text intended to obtain IOCs from tweets.
+First substitutions are performed and then words in the text are compared with
+'(/|^)([a-z0-9.-]+\.[a-z0-9]+?)([/:]|$)'
+In the case of a match it is checked whether this can be a valid domain using get_tld
+There is also a whitelist for filtering out good domains.
+
+
+Parameters:
+    domain_whitelist : domains that will be ignored in parsing
+    substitutions : semicolon separated list of pairs
+                    substitutions that will be made in the text,
+                    for example " .com,.com" enables parsing of one fuzzy format
+                                "[.];." enables the parsing of another fuzzy format
+
+"""
+
+from url_normalize import url_normalize
+from tld import get_tld
+from tld.utils import update_tld_names
+from intelmq.lib.bot import ParserBot
+from intelmq.lib.bot import utils
+import re
+
+
+class TwitterParserBot(ParserBot):
+    def init(self):
+        update_tld_names()
+        self.domain_whitelist = []
+        if getattr(self.parameters, "domain_whitelist", '') is not '':
+            self.domain_whitelist.extend(self.parameters.domain_whitelist.split(','))
+        self.substitutions = []
+        if getattr(self.parameters, "substitutions", '') is not '':
+            temp = self.parameters.substitutions.split(';')
+            if len(temp) % 2 != 0:
+                raise exceptions.InvalidArgument(
+                        'substitutions',
+                        got=self.parameters.substitutions,
+                        expected="even number of strings")
+            for i in range(int(len(temp)/2)):
+                self.substitutions.append([temp[2*i], temp[2*i+1]])
+
+    def get_domain(self, address):
+        try:
+            dom = re.search('(/|^)([a-z0-9.-]+\.[a-z0-9]+?)([/:]|$)', address).group(2)
+            if not self.in_whitelist(dom):
+                if get_tld(url_normalize(dom), fail_silently=True):
+                    return url_normalize(dom)
+                else:
+                    return None
+        except:
```

Comment:
do not use bare excepts, use at least `except Exception` if you do not know the exact exception you are expecting here (I think you want `TypeError`)